collect_app/src/main/java/org/odk/collect/android/formmanagement/FormUpdateChecker.kt

```diff
@@ -0,0 +1,100 @@
+package org.odk.collect.android.formmanagement
+
+import android.content.Context
+import org.javarosa.core.reference.ReferenceManager
+import org.odk.collect.analytics.Analytics
+import org.odk.collect.android.R
+import org.odk.collect.android.backgroundwork.ChangeLock
+import org.odk.collect.android.notifications.Notifier
+import org.odk.collect.android.preferences.keys.GeneralKeys
+import org.odk.collect.android.preferences.source.SettingsProvider
+import org.odk.collect.android.provider.FormsProviderAPI
+import org.odk.collect.android.storage.StoragePathProvider
+import org.odk.collect.android.storage.StorageSubdirectory
+import org.odk.collect.android.utilities.FormsDirDiskFormsSynchronizer
+import org.odk.collect.android.utilities.FormsRepositoryProvider
+import org.odk.collect.android.utilities.TranslationHandler
+import org.odk.collect.forms.FormSourceException
+import java.io.File
+import java.util.stream.Collectors
+
+class FormUpdateChecker(
+    private val context: Context,
+    private val notifier: Notifier,
+    private val analytics: Analytics,
+    private val changeLock: ChangeLock,
+    private val storagePathProvider: StoragePathProvider,
+    private val settingsProvider: SettingsProvider,
+    private val formsRepositoryProvider: FormsRepositoryProvider,
+    private val formSourceProvider: FormSourceProvider
+) {
+
+    fun checkForUpdates(projectId: String): Boolean {
```

Comment:
This name is kind of confusing taking into account that this method might just check for updates but also download them. A method that is called like that shouldn't do anything more than just checking if updates are available.
I think it would be better to split it into smaller use cases like `checkForUpdates` and `downloadUpdates` and use them depending on our settings not like now ran the job and then at the last moment decide what to do. Did you think about it?

Reply:
The problem is that they're not additive - we don't check for update **then** download updates, we do one or the other. We could just have two different methods and call different ones from `AutoUpdateTaskSpec` based on the setting, but I'd like to have that decision outside an "Android" component and I don't personally feel like it makes sense to add another layer.

I do think the naming is a bit misleading though (although there are tests for the method that describe the behaviour). I'm thinking I'll rename it `downloadUpdates` (as that's the "primary" use case) and add some JavaDoc.